x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/analyses/Classification.java

```diff
@@ -108,19 +120,26 @@ public Classification(String dependentVariable,
         this.dependentVariable = ExceptionsHelper.requireNonNull(dependentVariable, DEPENDENT_VARIABLE);
         this.boostedTreeParams = ExceptionsHelper.requireNonNull(boostedTreeParams, BoostedTreeParams.NAME);
         this.predictionFieldName = predictionFieldName == null ? dependentVariable + "_prediction" : predictionFieldName;
+        this.classAssignmentObjective = classAssignmentObjective == null ?
+            ClassAssignmentObjective.MAXIMIZE_MINIMUM_RECALL : classAssignmentObjective;
         this.numTopClasses = numTopClasses == null ? DEFAULT_NUM_TOP_CLASSES : numTopClasses;
         this.trainingPercent = trainingPercent == null ? 100.0 : trainingPercent;
         this.randomizeSeed = randomizeSeed == null ? Randomness.get().nextLong() : randomizeSeed;
     }
 
     public Classification(String dependentVariable) {
-        this(dependentVariable, BoostedTreeParams.builder().build(), null, null, null, null);
+        this(dependentVariable, BoostedTreeParams.builder().build(), null, null, null, null, null);
     }
 
     public Classification(StreamInput in) throws IOException {
         dependentVariable = in.readString();
         boostedTreeParams = new BoostedTreeParams(in);
         predictionFieldName = in.readOptionalString();
+        if (in.getVersion().onOrAfter(Version.V_7_7_0)) {
```

Comment:
Both this and the one in the write function will have to be changed to version 8.0.0 initially. We'll change them back to 7.7.0 after backporting to the 7.x branch.

Reply:
Are right, for backwards compatibility tests?

Reply:
Yes. In fact, it'd also be great to add the parameter in the classification BWC tests. I'll ping you to show you where they are.

Reply:
I added the tests in [d635bde](https://github.com/elastic/elasticsearch/pull/52763/commits/d635bde1762e1cbff76d1a95dd168b26807f10b0) and temporarily switched the version to 8.0 in [a1b5e2b](https://github.com/elastic/elasticsearch/pull/52763/commits/a1b5e2b90b30d20cebbf9febe8a78be8f73e9404).